fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java

```diff
@@ -35,7 +35,7 @@ private SQLInjectionValidator() {
 
     private static final String[] COMMENTS = { "--", "({", "/*", "#" };
 
-    private static final String SQL_PATTERN = "[a-zA-Z_=,\\-'!><.?\"`% ()0-9*\n\r]*";
+    private static final String SQL_PATTERN = "[a-zA-Z_=,\\-:'!><.?\"`% ()0-9*\n\r]*";
```

Comment:
One thing here. If you really wanna touch this class then cover these with tests. We literally have zero coverage on this and we could easily break it.

Reply:
I don't want to touch it, the function needs it. If you want to search for a date like this: 2100-12-12 20:20:20. The current implementation will treat it as SQL injection however it is not. I will write some testcases for it.

Reply:
Yeah I got it you didn't want to touch it but the thing is you did hence the testing is on you. Please add proper test cases to cover this functionality (not only for your date case).